### DISCUSSION TITLE
How do I get quart and discord.py to share the same event loop?

### DISCUSSION THREAD

**Discussion by dyc3** [Rapptz/discord.py#6519] [Q&A]
I've been trying to get a flask server and discord.py to play nice with each other. I thought making it all async would help, so I tried using quart instead.

My current code roughly looks like this.

```python3
bot = commands.Bot(command_prefix='>')

await asyncio.gather(serve.app.run_task(host="0.0.0.0", use_reloader=False), bot.start(get_token()))
```

When I do this, the quart server reponds, but the discord bot doesn't. How do I get discord.py to share the event loop with quart?

I can share more code if needed, I'm really sleepy rn

**Comment by mesub7** [comment]
Hi there,

You can use an unofficial extension called `discord-ext_ipc`. It's an IPC extension which allows you to transfer data from your bot to a separate process.

The repo is here: https://github.com/ext-creators/discord-ext-ipc
An example (which uses Quart like you): https://github.com/Ext-Creators/discord-ext-ipc/tree/master/examples
Documentation: https://discord-ext-ipc.readthedocs.io/en/latest/index.html

I must stress that **this is an unofficial extension**, so support is unlikely to be given in the discord.py server (there's a server for this in the repo's readme).

Hopefully this is what you're after.
